File: frontend/src/app/models/internship.ts
import { Application } from './application';

/** Matches the backend Internship entity as serialised by the API. */
export interface Internship {
  id: number;
  title: string;
  description: string | null;
  company: string;
  duration: string | null;
  location: string | null;
  /** ISO 8601 date-time string. */
  postedDate: string;
  applications?: Application[];
}

/** Shape accepted by POST /api/Internships and PUT /api/Internships/{id}. */
export interface InternshipRequest {
  title: string;
  description?: string | null;
  company: string;
  duration?: string | null;
  location?: string | null;
}

File: backend/Models/Entities/Internship.cs
using System.ComponentModel.DataAnnotations;

namespace backend.Models.Entities;

public class Internship
{
    public int Id { get; set; }

    [Required]
    [MaxLength(200)]
    public string Title { get; set; } = string.Empty;

    [MaxLength(2000)]
    public string? Description { get; set; }

    [Required]
    [MaxLength(150)]
    public string Company { get; set; } = string.Empty;

    [MaxLength(100)]
    public string? Duration { get; set; }

    [MaxLength(150)]
    public string? Location { get; set; }

    public DateTime PostedDate { get; set; } = DateTime.UtcNow;

    public ICollection<Application> Applications { get; set; } = new List<Application>();
}
